front/lib/actions/mcp_internal_actions/servers/google_calendar.ts

```diff
@@ -424,4 +540,46 @@ const createServer = (): McpServer => {
   return server;
 };
 
+function enrichEventWithDayOfWeek(
+  event: GoogleCalendarEvent
+): EnrichedGoogleCalendarEvent {
+  if (!event) {
+    return event;
+  }
+
+  const enrichedEvent: EnrichedGoogleCalendarEvent = { ...event };
+
+  if (event.start?.dateTime) {
+    const startDate = new Date(event.start.dateTime);
+    enrichedEvent.start = {
+      ...event.start,
+      dayOfWeek: startDate.toLocaleDateString("en-US", { weekday: "long" }),
+    };
+  } else if (event.start?.date) {
+    // Handle all-day events
+    const startDate = new Date(event.start.date);
+    enrichedEvent.start = {
+      ...event.start,
+      dayOfWeek: startDate.toLocaleDateString("en-US", { weekday: "long" }),
+    };
+  }
+
+  if (event.end?.dateTime) {
+    const endDate = new Date(event.end.dateTime);
+    enrichedEvent.end = {
+      ...event.end,
+      dayOfWeek: endDate.toLocaleDateString("en-US", { weekday: "long" }),
+    };
+  } else if (event.end?.date) {
+    // Handle all-day events
+    const endDate = new Date(event.end.date);
+    enrichedEvent.end = {
+      ...event.end,
+      dayOfWeek: endDate.toLocaleDateString("en-US", { weekday: "long" }),
```

Comment:
We have to do everything with the locale of the user here (which we have).

I also think all day events should be shown as such to the model?